examples/README.md

```diff
@@ -2,52 +2,132 @@
-
--   启动 AMOP 服务端：同单播案例
-
--   修改配置文件：同单播案例
-
--   运行 AMOP 客户端：
-
-    ```shell
-    go run examples/singleclient/unicast_client.go
-    ```
-
-    服务端控制台输出 "hello, multi broadcasts!"，客户端输出 "PushTopicDataToALL success" 表示运行成功
+**多播** 指的时节点向监听同一个 Topic 的所有订阅者转发消息。只要网络正常，即使没有监听 Topic 的订阅者，消息发布者也会收到消息发送成功的网络回包
+
+- 运行 AMOP 消息发布者：go run examples/amop/multicast_pub/publisher.go [endpoint] [topic]
+
+  ```shell
+  > go run examples/amop/multicast_pub/publisher.go 127.0.0.1:20200 hello
+  
+  2020/08/10 22:54:06 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 0 
+  2020/08/10 22:54:08 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 1 
+  2020/08/10 22:54:10 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 2 
+  2020/08/10 22:54:12 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 3 
+  2020/08/10 22:54:14 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 4 
+  2020/08/10 22:54:16 publish message: hello, FISCO BCOS, I am multi broadcast publisher! 5
+  ```
+
+- 启动 AMOP 消息订阅者：go run examples/amop/sub/subscriber.go [endpoint] [topic]
+
+  ```shell
+  > go run examples/amop/sub/subscriber.go 127.0.0.1:20201 hello
+  
+  Subscriber success
+  2020/08/10 22:54:12 received: hello, FISCO BCOS, I am multi broadcast publisher! 3
+  2020/08/10 22:54:14 received: hello, FISCO BCOS, I am multi broadcast publisher! 4
+  2020/08/10 22:54:16 received: hello, FISCO BCOS, I am multi broadcast publisher! 5
+  ```
+
+## 身份验证单播案例
+
+在普通的配置下，任何一个监听了某topic的接收者都能接受到发送者推送的消息。但在某些场景下，发送者只希望特定的接收者能接收到消息，不希望无关的接收者能任意的监听此topic。在此场景下，需要使用Topic认证功能。 认证功能是指对于特定的topic消息，允许通过认证的接收者接收消息。详细请参考：[Topic认证功能](https://fisco-bcos-doc-qiubing.readthedocs.io/en/latest/docs/manual/amop_protocol.html#topic)
+
+- 运行 AMOP 消息发布者：go run examples/amop_auth/unicast_pub/publisher.go [endpoint] [topic]
+
+  ```shell
+  > go run examples/amop_auth/unicast_pub/publisher.go 127.0.0.1:20200 hello
+  
+  publish topic success
+  2020/08/10 23:18:50 publish message: Hi, FISCO BCOS! 0 
+  2020/08/10 23:18:52 PushAuthTopicDataRandom failed, err: sendMessage failed, err: error code 100, remote peer unavailable
+  2020/08/10 23:18:52 publish message: Hi, FISCO BCOS! 1 
+  2020/08/10 23:18:54 publish message: Hi, FISCO BCOS! 2 
+  2020/08/10 23:18:56 publish message: Hi, FISCO BCOS! 3 
+  2020/08/10 23:18:58 publish message: Hi, FISCO BCOS! 4
+  ```
+
+- 启动 AMOP 消息订阅者：go run examples/amop_auth/sub/subscriber.go [endpoint] [topic]
+
+  ```shell
+  > go run examples/amop_auth/sub/subscriber.go 127.0.0.1:20201 hello
+  
+  Subscriber success
+  2020/08/10 23:18:52 received: Hi, FISCO BCOS! 1
+  2020/08/10 23:18:54 received: Hi, FISCO BCOS! 2
+  2020/08/10 23:18:56 received: Hi, FISCO BCOS! 3
+  2020/08/10 23:18:58 received: Hi, FISCO BCOS! 4
+  ```
+
+## 身份验证多播案例
+
+- 运行 AMOP 消息发布者：go run examples/amop_auth/multicast_pub/publisher.go [endpoint] [topic]
```

Comment:
Codacy found an issue: [[no-undefined-references] Found reference to undefined definition](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5979849)